### PR TITLE
Add `dir` and `textAlign` in `FormatState`

### PR DESCRIPTION
**Pull request by Echo-az**
Add dir and text-align info in FormatState. They will be used to calculate the alignment (left | center | right) of text at cursor. This state will be displayed to the users through the icons of the button.
image

### CODE REVIEW

**Comment by Echo-az** [comment]
@microsoft-github-policy-service agree company="Microsoft"

File: packages/roosterjs-editor-types/lib/interface/FormatState.ts [hunk 1]
```diff
@@ -103,6 +103,11 @@ export interface ElementBasedFormatState {
      * Whether we can execute table cell merge operation
      */
     canMergeTableCell?: boolean;
+
+    /**
+     * Dir of the element ('ltr' or 'rtl')
+     */
+    dir?: string;
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-types/lib/interface/FormatState.ts:110]: This should be part of StyleBasedFormatState

File: packages/roosterjs-editor-api/lib/format/getFormatState.ts [hunk 2]
```diff
@@ -42,6 +42,13 @@ export function getElementBasedFormatState(
         ? toArray(table.rows[0].cells).every(cell => getTagOfNode(cell) == 'TH')
         : undefined;
 
+    let dir = undefined;
+    let element = editor.getElementAtCursor();
+    while (!dir && element) {
+        dir = element.getAttribute('dir');
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-api/lib/format/getFormatState.ts:48]: dir can also be specified using CSS "direction".

So do this in getStyleBasedFormatState is much easier.

**Comment by JiuqingSong** [comment]
Can you also modify https://github.com/microsoft/roosterjs/blob/1884e38d621cd9612f6a37a3dcec26b837e8274e/packages/roosterjs-content-model/lib/modelApi/common/retrieveModelFormatState.ts#L88 to let the new retrieveModelFormatState function also return correct value of your new attributes? This is a nice-to-have.

File: packages/roosterjs-editor-api/lib/format/getFormatState.ts (hunk 2 above)
Comment (Echo-az) [reply packages/roosterjs-editor-api/lib/format/getFormatState.ts]: Thanks for suggestion! Moved and renamed it.

**Review by JiuqingSong — APPROVED** [review]

**Comment by JiuqingSong** [comment]
@Echo-az  There are test cases fail, would you please fix them as well?
You can run `yarn test` to run test locally.

**Comment by Echo-az** [comment]
> @Echo-az There are test cases fail, would you please fix them as well? You can run `yarn test` to run test locally.

Hi @JiuqingSong, I ran `yarn test` both before and after my change. There're some failed test cases even without any change. But the count of failed test cases didn't increase, I'm not sure my change is ok. Do you have any idea about this?

> Chrome 110.0.0.0 (Mac OS 10.15.7): Executed 3455 of 3689 (12 FAILED) (skipped 234) (2.828 secs / 2.188 secs)
> Something in shortcutFeatureTest.ts, undoPluginTest.ts, HyperLinkTest.ts

**Review by JiuqingSong — APPROVED** [review]

**Pull request merged**